File: Projetos/ACBrLib/Demos/C#/Shared/ACBrLib.Core/Config/ACBrLibConfig.cs
namespace ACBrLib.Core.Config
{
    public abstract class ACBrLibConfig<TLib> : ACBrLibConfigBase<TLib> where TLib : ACBrLibHandle
    {
        #region Constructors

        protected ACBrLibConfig(TLib acbrlib, ACBrSessao sessao) : base(acbrlib, sessao)
        {
            Principal = new PrincipalConfig<TLib>(Parent);
            Sistema = new SistemaConfig<TLib>(Parent);
            Proxy = new ProxyConfig<TLib>(Parent);
            SoftwareHouse = new SoftwareHouseConfig<TLib>(Parent);
            Emissor = new EmissorConfig<TLib>(Parent);
        }

        #endregion Constructors

        #region Properties

        public PrincipalConfig<TLib> Principal { get; }

        public SistemaConfig<TLib> Sistema { get; }

        public ProxyConfig<TLib> Proxy { get; }

        public SoftwareHouseConfig<TLib> SoftwareHouse { get; }

        public EmissorConfig<TLib> Emissor { get; set; }

        #endregion Properties
    }
}

File: Projetos/ACBrLib/Demos/Java/ACBr Common/src/com/acbr/ACBrLibConfig.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.acbr;

/**
 *
 * @author rften
 * @param <TLib>
 */
public abstract class ACBrLibConfig<TLib extends ACBrLibBase> extends ACBrLibConfigBase<TLib> {
    
    private final PrincipalConfig<TLib> principal;
    private final SistemaConfig<TLib> sistema;
    
    public ACBrLibConfig(TLib acbrlib, ACBrSessao sessao) {
        super(acbrlib, sessao);
        
        principal = new PrincipalConfig<>(acbrlib);
        sistema = new SistemaConfig<>(acbrlib);
    }
    
    public PrincipalConfig<TLib> getPrincipal(){
        return principal;
    }
    
    public SistemaConfig<TLib> getSistema(){
        return sistema;
    }
    
}
